src/panels/lovelace/cards/hui-entities-card.ts

```diff
@@ -1,186 +1,182 @@
-    ) {
-      element.classList.add("state-card-dialog");
-      element.addEventListener("click", () => this._handleClick(entityConf));
-    }
-
-    return element;
-  }
-
-  private _handleClick(entityConf: ConfigEntity) {
-    const entityId = entityConf.entity;
-
-    fireEvent(this, "hass-more-info", { entityId });
-  }
-}
-
-declare global {
-  interface HTMLElementTagNameMap {
-    "hui-entities-card": HuiEntitiesCard;
-  }
-}
-
-customElements.define("hui-entities-card", HuiEntitiesCard);
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+
+import "../../../components/ha-card.js";
+import "../components/hui-entities-toggle.js";
+
+import { fireEvent } from "../../../common/dom/fire_event.js";
+import { DOMAINS_HIDE_MORE_INFO } from "../../../common/const.js";
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import createRowElement from "../common/create-row-element.js";
+import computeDomain from "../../../common/entity/compute_domain.js";
+import processConfigEntities from "../common/process-config-entities";
+import { HomeAssistant } from "../../../types.js";
+import { EntityConfig, EntityRow } from "../entity-rows/types.js";
+
+interface ConfigEntity extends EntityConfig {
+  type?: string;
+  secondary_info: "entity-id" | "last-changed";
+  action_name?: string;
+  service?: string;
+  service_data?: object;
+  url?: string;
+}
+
+interface Config extends LovelaceConfig {
+  show_header_toggle?: boolean;
+  title?: string;
+  entities: ConfigEntity[];
+}
+
+class HuiEntitiesCard extends hassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  protected _hass?: HomeAssistant;
+  protected _config?: Config;
+  protected _configEntities?: ConfigEntity[];
+
+  set hass(hass) {
+    this._hass = hass;
+    this.shadowRoot!.querySelectorAll("#states > *").forEach(
+      (element: unknown) => {
+        (element as EntityRow).hass = hass;
+      }
+    );
+  }
+
+  static get properties(): PropertyDeclarations {
+    return {
+      _config: {},
+    };
+  }
+
+  public getCardSize() {
+    if (!this._config) {
+      return 0;
+    }
+    // +1 for the header
+    return (this._config.title ? 1 : 0) + this._config.entities.length;
+  }
+
+  public setConfig(config: Config) {
+    const entities = processConfigEntities(config.entities);
+    for (const entity of entities) {
+      if (
+        entity.type === "call-service" &&
+        (!entity.service || !entity.name || !entity.icon || !entity.action_name)
+      ) {
+        throw new Error("Missing required property when type is call-service");
+      } else if (
+        entity.type === "weblink" &&
+        (!entity.name || !entity.icon || !entity.url)
+      ) {
+        throw new Error("Missing required property when type is weblink");
+      }
+    }
+
+    this._config = config;
+    this._configEntities = entities;

```

Comment:
We should call it `_configRows`, as not all are entities.

Reply:
I think prettier ran against this file and made it look like I touched everything. Fail on my part.

My only intended change was to remove the `!entity.service_data` check when `entity.type === "call-service"` as I moved that to being optional. The full check should be removed in a future PR